src/login/logind-dbus.c

```diff
@@ -1488,59 +1493,32 @@ static int have_multiple_sessions(
         return false;
 }
 
-_printf_(2, 0)
-static int log_with_wall_message(Manager *m, const char *d, const char *p, const char *q) {
+static int bus_manager_log_shutdown(
+                Manager *m,
+                const ActionTableItem *a) {
+
+        const char *message = a->message;
+        const char *log_str = a->log_str;
+
         assert(m);
 
-        if (isempty(m->wall_message))
-                p = strjoina(p, ".");
+        if (message)
+                message = strjoina("MESSAGE=", message);
         else
-                p = strjoina(p, " (", m->wall_message, ").");
+                message = "MESSAGE=System is shutting down";
 
-        return log_struct(LOG_NOTICE, d, p, q);
-}
+        if (isempty(m->wall_message))
+                message = strjoina(message, ".");
+        else
+                message = strjoina(message, " (", m->wall_message, ").");
```

Comment:
Hmm, this is a user-controlled message (via polkit). So I don't think we should use `strjoina` here, but normal `strjoin` instead. This is not a very pressing issue, but it's a bit sloppy.